targetselector/running_target_selector.go

```diff
@@ -75,6 +75,15 @@ func (r runningTargetSelector) GetMatchingPodsOverTotalPods(c client.Client, ins
 			}
 		}
 
+		if instance.Spec.Filters != nil && len(instance.Spec.Filters.Annotations) > 0 {
+			for k, v := range instance.Spec.Filters.Annotations {
+				podAnno, ok := pod.Annotations[k]
+				if !ok || podAnno != v {
+					continue
+				}
+			}
+		}
```

Comment:
```suggestion
                 // apply filters on matching resources
		if instance.Spec.Filters {
		        // apply annotation filters
			for k, v := range instance.Spec.Filters.Annotations {
				podAnno, ok := pod.Annotations[k]
				if !ok || podAnno != v {
					continue
				}
			}
		}
```

The `for range` clause will be skipped if annotations is empty so it's not needed to check its length + it prepares the ground if we want to add other kind of filters later on.